lib/dash/segment_list.js

```diff
@@ -90,6 +92,40 @@ shaka.dash.SegmentList = class {
     };
   }
 
+  /**
+   * Gets timescale from SegmentList.
+   *
+   * @param {shaka.dash.DashParser.Context} context
+   * @return {number}
+   * @private
+   */
+  static getTimescaleFromSegmentList_(context) {
+    const Functional = shaka.util.Functional;
+    /** @type {!Array.<!Element>} */
+    const segmentLists = [
+      context.representation.segmentList,
+      context.adaptationSet.segmentList,
+      context.period.segmentList,
+    ].filter(Functional.isNotNull);
+
+    if (!segmentLists.length) {
+      return 1;
+    }
+
+    const XmlUtils = shaka.util.XmlUtils;
+    const node = XmlUtils.findChildren(segmentLists[0], 'SegmentList');
+    if (!node) {
+      return 1;
+    }
+
+    const timescale = segmentLists[0].getAttribute('timescale');
```

Comment:
I have seen many instances where the DASH timescale disagreed with the MP4 box timescale.  In those cases, I think this would put a value in the event that would make it impossible to interpret mediaTime correctly.

Should we perhaps parse the init segment for the timescale instead?  See usage of `shaka.util.Mp4BoxParsers.parseMDHD`, for example in `lib/cea/mp4_cea_parser.js` or `lib/text/mp4_vtt_parser.js`.

If we do as I'm suggesting, the timescale value in InitSegmentReference would remain unset until the init segment was fetched by StreamingEngine.

Reply:
Good Idea. I will look into that solution.

Reply:
Hi @joeyparrish,

I have been testing with our live streams and I do not see the `MDHD` being present.
I do however see `timescale` present in the `SIDX`, however checking the docs `SIDX` does not appear to be mandatory.

What do you think about checking the `MDHD`, if it's not present checking for `SIDX` instead, if still not present then defaulting to what it says in the manifest?

<img width="762" alt="image" src="https://user-images.githubusercontent.com/15213588/183644999-f25b7100-4b0a-4a4b-8beb-d6a142109166.png">

Reply:
sidx timescale would refer to the media times in the sidx specifically.  So that is allowed to be different from what's in the mdhd header.  You should expect mdhd in the init segment, though, not the media segment.